components/ntask/org.wso2.carbon.ntask.core/src/main/java/org/wso2/carbon/ntask/core/impl/clustered/ClusterGroupCommunicator.java

```diff
@@ -78,28 +72,17 @@ public static ClusterGroupCommunicator getInstance(String taskType) throws TaskE
     private ClusterGroupCommunicator(String taskType) throws TaskException {
         this.taskType = taskType;
         this.taskService = TasksDSComponent.getTaskService();
-        this.hazelcast = TasksDSComponent.getHazelcastInstance();
-        if (this.getHazelcast() == null) {
-            throw new TaskException("ClusterGroupCommunicator cannot initialize, " +
+        throw new TaskException("ClusterGroupCommunicator cannot initialize, " +
             		"Hazelcast is not initialized", Code.CONFIG_ERROR);
```

Comment:
Are we throwing a TaskException if we hit the `ClusterGroupCommunicator` method? Don't we have any conditions that need to be checked here?

Reply:
I have been informed that Hazelcast usage is completely removed and I was asked to remove the missing code parts that use Hazelcast so we can completely remove the Hazelcast jar from the product. So technically in this case, this method cannot be run in any of our use-cases because if so, it's using Hazelcast from line 81 in old code. 

I wasn't able to remove all the unused methods completely because a lot of repositories and modules refer them even-though we actually didn't use them. So to remove Hazelcast with time concerns, followed the approach of removing them within methods. 

Since we have removed Hazelcast, throwing this without a condition because in all cases there will be no Hazelcast.